gapic/schema/wrappers.py

```diff
@@ -230,6 +234,20 @@ def ident(self) -> metadata.Address:
         """Return the identifier data to be used in templates."""
         return self.meta.address
 
+    @property
+    def resource_path(self) -> Optional[str]:
+        """If this message describes a resource, return the path to the resource.
+        If there are multiple paths, returns the first one."""
```

Comment:
How about other patterns? Do we not generate helpers for them?

Reply:
What other patterns does the client know or care about? A cursory perusal of the monolithic templates leads me to the (tentative) conclusion that the helper methods are for describing paths in request fields.

Reply:
This falls into the multi-pattern resource name black hole...

Gapic yaml has an entity name for every resource pattern, and the entity name is used to generate the name of the helper method.

In annotations however, we only have an identifier for every resource. For single-pattern resources we are fine because the resource is the pattern. However, for multi-pattern resources the generators don't know what the resource name helpers for each pattern should be named because we don't have that information.

Some design work might be needed for multi-pattern resources in Python. I was under the impression of discontinuing the resource name helpers in the micro generator so I kinda skipped the design of that in Python. It might be reasonable to take the same approach as Ruby and Node do, described in https://docs.google.com/document/d/1nuh6yrPubl1XMWH2WQHtmOwk_s-0MZEgSoFD6RbMrxg/edit#heading=h.a3gbsu9p63o

Reply:
Ahh, multipattern resources. Got it.

The consensus was to take a breaking change and not support multipattern resources. It's just easier not to :P

Reply:
... well okay then.

Reply:
Dynamic languages make the issue worse because there isn't a good mechanism to have method overloading. Without method overloading, we would need some kind of name disambiguation, and moving away from the gapic config makes that more difficult. Given the small number of multi-pattern resources, the consensus was "screw it."